src/app/api/index.ts

```diff
@@ -319,44 +322,82 @@ class Client implements API {
         return newAssignment;
     }
 
-    async createAssignmentTemplate(newTemplate: SheriffAssignmentTemplate): Promise<SheriffAssignmentTemplate> {
-        await randomDelay();
-        let assignment = newTemplate.assignmentTemplate;
-
-        //This is a hack to create a unique id for a new assignment template
-        newTemplate.id = defaultAssignmentTemplates.length;
-        assignment.id = defaultAssignmentTemplates.length;        
+    private setAssignmentTemplateTitle(assignment: Partial<SheriffAssignment>): string {
 
-        // set the assignment template title
         if (assignment.workSectionId) {
-            if (WORK_SECTIONS[assignment.workSectionId] === WORK_SECTIONS.COURTS && assignment.title) {
-                newTemplate.assignmentTemplate.title = COURTROOMS[assignment.title];
+            if (WORK_SECTIONS[assignment.workSectionId] === WORK_SECTIONS.COURTS && assignment.courtroomId) {
+                return COURTROOMS[assignment.courtroomId];
             }
             else {
-                newTemplate.assignmentTemplate.title = WORK_SECTIONS[assignment.workSectionId];
+                return WORK_SECTIONS[assignment.workSectionId];
             }
         }
+        else {
+            return "Assignment Template";
+        }
 
-        //add default recurrence value if nothing was selected or partial value was selected
-        if(!newTemplate.recurrenceInfo){
-            newTemplate.recurrenceInfo = [DEFAULT_RECURRENCE];
+    }
+
+    private setAssignmentTemplateDefaultRecurrenceInfo(template: Partial<SheriffAssignmentTemplate>): RecurrenceInfo[] {
```

Comment:
in this case, this method is mutating it's input.  Sometimes this can lead to less transparent code than providing pure methods that take an input and return something.  For instance:

```
 setAssignmentTemplateDefaultRecurrenceInfo(template)
// VERSUS
template.recurrenceInfo = createFilledRecurrenceInfo(template.recurrenceInfo);
```
It's very explicit that we are replacing the recurrenceInfo here with a filledVersion

Where your method could look like:
```ts

private createFilledRecurrenceInfo(recurrenceInfo:RecurrenceInfo[]):RecurrenceInfo[]{
  // Here we take the existing recurrenceInfo and return a new array by mapping each one
  // to a new object that is merged over top of the DEFAULT.  This means, that any key
  // that matches a key of DEFAULT_RECURRENCE will replace the value in the new object
  // Effectively doing the same thing you're doing below
 return recurrenceInfo.map(r=>Object.assign({},DEFAULT_RECURRENCE,r));
}

```